python/sglang/srt/layers/moe/fused_moe_triton/layer.py

```diff
@@ -690,9 +743,19 @@ def forward(self, hidden_states: torch.Tensor, router_logits: torch.Tensor):
             activation=self.activation,
             apply_router_weight_on_input=self.apply_router_weight_on_input,
             routed_scaling_factor=self.routed_scaling_factor,
+            **(
+                dict(
+                    tp_rank=self.tp_rank,
+                    tp_size=self.tp_size,
+                    ep_rank=self.ep_rank,
+                    ep_size=self.ep_size,
+                )
+                if self.quant_method.__class__.__name__ == "ModelOptNvFp4FusedMoEMethod"
+                else {}
+            ),
         )
 
-        if self.reduce_results and self.tp_size > 1:
+        if self.reduce_results and (self.tp_size > 1 or self.ep_size > 1):
```

Comment:
I think if DP attention is on, reduce-scatter is sufficient here instead of all-reduce.

Reply:
Does this PR work naturely for DP-attention? I think you may consider quantization -> collective->fused_moe -> collective pattern.